### ISSUE TITLE
Pipelining is missing

### ISSUE THREAD

**Issue by Asmod4n** [matsumoto-r/mruby-redis#46]
So far, Redis pipelining is missing from this project.

There are at least two ways to add it:
- create a special proxy class which internally stores a list of commands and pipelines them to the Redis server.
```ruby
redis.pipeline do |pipe|
  pipe.set("key", "value")
  pipe.incr("number")
end # would give a array of results back for each command
```
 the downside here is: there would either reside two "client classes" in here, which both do mainly the same and someone would have to maintain both, or it would be based upon ```method_missing```.

- take over the code i have written at https://github.com/Asmod4n/mruby-hiredis for pipelining and adopt it a bit for this project
```ruby
redis.queue(:set, "key", "value")
redis.queue(:incr, "number")
redis.queue_end # would give a array of results back for each command
```

The original Ruby client for Redis does all kind of "magic" to make it feel like ruby. But the goal of pipelining is increased speed, so i got mixed feelings about it.

**Comment by matsumoto-r** [comment]
Thank you for your proposal. I like `redis.queue(:set, "key", "value")`.

**Comment by Asmod4n** [comment]
closed via #49
